commit ddb8a8a022b9733e50a353d1d09a95fcdb8d2b8f
Author: Ye Li <ye.li@nxp.com>
Date:   Wed Oct 16 16:31:13 2024 -0400

    EDAC/fsl_ddr: Add support for i.MX9 DDR controller
    
    Add support for the i.MX9 DDR controller, which has different register
    offsets and some function changes compared to the existing fsl_ddr
    controller. The ECC and error injection functions are almost the same,
    so update and reuse the driver for i.MX9. Add a special type 'TYPE_IMX9'
    specifically for the i.MX9 controller to distinguish the differences.
    
    Signed-off-by: Ye Li <ye.li@nxp.com>
    Signed-off-by: Frank Li <Frank.Li@nxp.com>
    Signed-off-by: Borislav Petkov (AMD) <bp@alien8.de>
    Reviewed-by: Peng Fan <peng.fan@nxp.com>
    Link: https://lore.kernel.org/r/20241016-imx95_edac-v3-5-86ae6fc2756a@nxp.com

diff --git a/drivers/edac/layerscape_edac.c b/drivers/edac/layerscape_edac.c
index 0d42c1238908..9a0c92ebbc3c 100644
--- a/drivers/edac/layerscape_edac.c
+++ b/drivers/edac/layerscape_edac.c
@@ -21,6 +21,7 @@
 
 static const struct of_device_id fsl_ddr_mc_err_of_match[] = {
 	{ .compatible = "fsl,qoriq-memory-controller", },
+	{ .compatible = "nxp,imx9-memory-controller", .data = (void *)TYPE_IMX9, },
 	{},
 };
 MODULE_DEVICE_TABLE(of, fsl_ddr_mc_err_of_match);